integration/unplugin/src/index.ts

```diff
@@ -363,7 +367,13 @@ export const rawPlugin: Parameters<typeof createUnplugin<PluginOptions>>[0] =
       const basename = id.slice(0, match.index + match[1].length);
 
       const filename = path.resolve(rootDir, basename);
-      const rawCivetSource = await fs.promises.readFile(filename, 'utf-8');
+      const mtime = fs.statSync(filename).mtimeMs;
+      const cached = cache?.get(filename);
+      if (cached && cached.mtime === mtime) {
+        return cached.result;
+      }
+
+      const rawCivetSource = fs.readFileSync(filename, 'utf-8');
```

Comment:
Is there a reason to switch to sync interface? I wonder if promises interface (also stat) might support multitasking slightly better.

Reply:
My intuition is that opening a lot of files and reading a little bit of each is slower than opening one and reading all of it but I haven't done a full benchmark yet.

Reply:
Ah, fair point - I'm not sure what's best. Probably fine either way.

Reply:
I did a brief unscientific test and the promises version seemed to finish faster. The promises version probably has better throughput where node/the os can buffer multiple files in parallel.